### PR TITLE
Fixes #158 - Adds Chrome debugging to ReactNative for UWP

### LINKED ISSUE #158: Add support for remote JavaScript debugging.

### PR DESCRIPTION
**Pull request by rozele**
This was a pretty trivial setup, given all the infrastructure and designs that were already in place to implement the DevSupportManager.

All we really needed was to investigate the correct format for data on the wire, and implement it.

### CODE REVIEW

File: ReactWindows/ReactNative/Bridge/WebSocketJavaScriptExecutor.cs [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+using System.Runtime.ExceptionServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Windows.Foundation;
+using Windows.Networking.Sockets;
+using Windows.Storage.Streams;
+
+namespace ReactNative.Bridge
+{
+    class WebSocketJavaScriptExecutor : IJavaScriptExecutor
```
Comment (rozele) [inline ReactWindows/ReactNative/Bridge/WebSocketJavaScriptExecutor.cs]: Consider making public and adding comments.

File: ReactWindows/ReactNative/DevSupport/DevServerHelper.cs [hunk 2]
```diff
@@ -221,5 +267,10 @@ private string CreateOnChangeEndpointUrl(string host)
         {
             return string.Format(CultureInfo.InvariantCulture, OnChangeEndpointUrlFormat, host);
         }
+
+        private string CreateLaunchDevToolsCommandUrl()
+        {
+            return string.Format(CultureInfo.InvariantCulture, LaunchDevToolsCommandUrlFormat, DebugServerHost);
```
Comment (rozele) [inline ReactWindows/ReactNative/DevSupport/DevServerHelper.cs]: Maybe spread this out with some newlines.

File: ReactWindows/ReactNative/DevSupport/DevSupportManager.cs [hunk 3]
```diff
@@ -121,7 +130,12 @@ public void HandleException(Exception exception)
             var progressDialog = new ProgressDialog("Please wait...", "Fetching JavaScript bundle.");
             var dialogOperation = progressDialog.ShowAsync();
 
-            if (_jsBundleFile == null)
+            if (_isUsingJsProxy)
+            {
+                await ReloadJavaScriptInProxyMode(progressDialog.Token);
```
Comment (rozele) [inline ReactWindows/ReactNative/DevSupport/DevSupportManager.cs]: Determine why progress dialog can end prematurely (how is the async context being lost?)

File: ReactWindows/ReactNative/DevSupport/DevServerHelper.cs [hunk 4]
```diff
@@ -86,6 +88,25 @@ private bool IsHotModuleReplacementEnabled
...
+        /// </summary>
+        public string WebsocketProxyUrl
+        {
+            get
+            {
+                return string.Format(CultureInfo.InvariantCulture, WebsocketProxyUrlFormat, DebugServerHost);
+            }
+        }
+
+        private string JavaScriptProxyHost
```
Comment (ebragge) [inline ReactWindows/ReactNative/DevSupport/DevServerHelper.cs]: Comment missing.

File: ReactWindows/ReactNative/DevSupport/DevServerHelper.cs [hunk 5]
```diff
@@ -221,5 +268,13 @@ private string CreateOnChangeEndpointUrl(string host)
         {
             return string.Format(CultureInfo.InvariantCulture, OnChangeEndpointUrlFormat, host);
         }
+
+        private string CreateLaunchDevToolsCommandUrl()
```
Comment (ebragge) [inline ReactWindows/ReactNative/DevSupport/DevServerHelper.cs]: Comment missing.

**Comment by ebragge** [comment]
:+1:

File: ReactWindows/ReactNative/DevSupport/DevServerHelper.cs (hunk 4 above)
Comment (rozele) [inline ReactWindows/ReactNative/DevSupport/DevServerHelper.cs]: It's a private property that will never become public.  

The only time I advocate for comments on non-public surfaces is when there is partial comment coverage of the public APIs on an internal class.  The reason I advocate for this is that it is sometimes the case that we need to refactor libraries and make internal classes public.  It's more of a policy of do everything, or nothing at all :-).

---
In reply to: [55510578](https://github.com/CatalystCode/react-native/pull/235#discussion_r55510578) [](ancestors = 55510578)

File: ReactWindows/ReactNative/DevSupport/DevServerHelper.cs (hunk 5 above)
Comment (rozele) [inline ReactWindows/ReactNative/DevSupport/DevServerHelper.cs]: Will not become public.

---
In reply to: [55510803](https://github.com/CatalystCode/react-native/pull/235#discussion_r55510803) [](ancestors = 55510803)

**Pull request merged**